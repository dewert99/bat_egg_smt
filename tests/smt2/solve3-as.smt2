(declare-sort S1)
(declare-const a S1)
(declare-fun b () S1)
(declare-const x Bool)
(assert x)
(assert ((as = Bool) (as a S1) (as b S1)))
(check-sat)
(get-value (a b x))
(get-model)
(check-sat-assuming (((as not Bool) (and x (= ((as a S1)) (b))))))
(get-unsat-core)