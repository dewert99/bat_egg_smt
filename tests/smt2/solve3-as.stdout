sat
((a S1!val!2)
 (b S1!val!2)
 (x true))
(
 (define-fun a () S1 S1!val!2)
 (define-fun b () S1 S1!val!2)
 (define-fun x () Bool true)
)
unsat
(((as not Bool) (and x (= ((as a S1)) (b)))))
